hack/generator/pkg/astmodel/armconversion/convert_to_arm_function_builder.go

```diff
@@ -66,6 +66,7 @@ func newConvertToARMFunctionBuilder(
 		result.fixedValuePropertyHandler("Type"),
 		result.fixedValuePropertyHandler("ApiVersion"),
 		result.propertiesWithSameNameHandler,
+		result.flattenedPropertyHandler,
```

Comment:
minor: order matters here I think, and you want to be above `propertiesWithSameNameHandler` I'm pretty sure

Reply:
Like you were in the `from_arm` direction

Reply:
So originally I was thinking this is a performance optimization, since the code must iterate to find all the flattened properties. However, there's a correctness issue here, when a flattened property contains a property that has the same name! So yes, this should come first. I'll also need a good test for it but will delay until golden files can support flattening properly…

Reply:
Tracking: https://github.com/Azure/azure-service-operator/issues/1597